src/components/Tooltip/index.tsx

```diff
@@ -1,147 +1,262 @@
+        arrowTop: tooltipHeight,
+        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
+        rotate: 0,
+        top: childrenTop - tooltipHeight - ARROW_WIDTH - SPACE,
+        translate: '',
+      }
+  }
+}
+
 type TooltipProps = {
-  animated?: number | boolean
-  baseId?: string
+  id?: string
   children: ReactNode
   maxWidth?: number
   placement?: TooltipPlacement
   text?: ReactNode
-  variant?: keyof typeof variants
-  visible?: boolean
-  unstable_portal?: boolean
-} & BoxProps
+  className?: string
+}
 
 const Tooltip = ({
-  animated = 150,
   children,
   text = '',
   placement = 'top',
-  visible = false,
-  variant = 'black',
-  baseId,
-  unstable_portal = true,
-  ...props
+  id,
+  className,
+  maxWidth = 232,
 }: TooltipProps): JSX.Element => {
-  const tooltip = useTooltipState({
-    animated,
-    baseId,
-    placement,
-    visible,
-  })
+  const childrenRef = useRef<HTMLDivElement>(null)
+  const tooltipRef = useRef<HTMLDivElement>(null)
+  const timer = useRef<ReturnType<typeof setInterval>>()
+  const [visibleInDom, setVisibleInDom] = useState(false)
+  const [positions, setPositions] = useState(DEFAULT_POSITIONS)
+
+  const computedId = useMemo(() => {
+    if (!id) return getUUID('tooltip')
+
+    return id
+  }, [id])
+
+  const getPositions = useCallback(() => {
+    if (childrenRef.current && tooltipRef.current) {
+      setPositions(computePositions({ childrenRef, placement, tooltipRef }))
+    }
+  }, [placement])
+
+  /**
+   * When mouse hover or stop hovering children this function display or hide tooltip. A timeout is set to allow animation
+   * end, then remove tooltip from dom.
+   */
+  const onMouseEvent = useCallback(
+    (isVisible: boolean) => () => {
+      // This is when we hide the tooltip, first we set opacity to 0 then we set a timeout based on CSS animation duration
+      // then we remove it from dom
+      if (!isVisible && tooltipRef.current) {
+        tooltipRef.current.style.opacity = '0'
+        timer.current = setTimeout(() => {
+          setVisibleInDom(isVisible)
+        }, ANIMATION_DURATION)
+      } else {
+        // If a timeout is already set it means tooltip didn't have time to close completely and be removed from dom,
+        // so we clear timeout and set back opacity of tooltip to 1, so it can be visible on screen.
+        if (timer.current) {
+          clearTimeout(timer.current)
+          if (tooltipRef.current) {
+            tooltipRef.current.style.opacity = '1'
+          }
+        }
+        setVisibleInDom(isVisible)
+      }
+    },
+    [],
+  )
 
-  const { setVisible } = tooltip
-  useEffect(() => setVisible(visible), [setVisible, visible])
+  /**
+   * Once tooltip is visible in the dom we can compute positions and then set it visible on screen.
+   */
+  useEffect(() => {
+    getPositions()
+
+    window.addEventListener('resize', getPositions)
+    window.addEventListener('scroll', getPositions)
```

Comment:
We are leaking memory by not removing those events in a clear callback

Reply:
@QuiiBz damn I forgot you're right!